arch/arm/core/aarch32/cortex_m/thread_abort.c

```diff
@@ -43,10 +41,12 @@ void z_impl_k_thread_abort(k_tid_t thread)
 			 */
 			SCB->ICSR |= SCB_ICSR_PENDSVSET_Msk;
 		} else {
-			z_swap_unlocked();
+			z_self_abort();
```

Comment:
@andrewboie do we need CODE_UNREACHABLE after this line, here?
Or at least a comment that indicates this, for ease of reading...

Reply:
@ioannisg `z_self_abort()` is `FUNC_NORETURN` but I added a `CODE_UNREACHABLE` line afterwards anyway for clarity.